apimatic_core/logger/configuration/api_logging_configuration.py

```diff
@@ -0,0 +1,131 @@
+import logging
+
+from apimatic_core_interfaces.logger.logger import Logger
+from apimatic_core.logger.default_logger import ConsoleLogger
+
+
+class ApiLoggingConfiguration:
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @property
+    def log_level(self):
+        return self._log_level
+
+    @property
+    def mask_sensitive_headers(self):
+        return self._mask_sensitive_headers
+
+    @property
+    def request_logging_config(self):
+        return self._request_logging_config
+
+    @property
+    def response_logging_config(self):
+        return self._response_logging_config
+
+    def __init__(self, logger, log_level, mask_sensitive_headers,
+                 request_logging_config, response_logging_config):
+        """Default constructor.
+
+        Args:
+            logger (Logger): The logger implementation to log with.
+            log_level (LogLevel): The log level to apply to the log message.
+            mask_sensitive_headers (bool): Flag to control masking of sensitive headers.
+            request_logging_config (ApiRequestLoggingConfiguration): The API request logging configuration.
+            response_logging_config (ApiResponseLoggingConfiguration): The API response logging configuration.
+        """
+
+        self._logger = ConsoleLogger() if logger is None else logger
+        self._log_level = logging.INFO if log_level is None else log_level
+        self._mask_sensitive_headers = mask_sensitive_headers
+        self._request_logging_config = request_logging_config
+        self._response_logging_config = response_logging_config
+
+
+class BaseHttpLoggingConfiguration:
+
+    @property
+    def log_body(self):
+        return self._log_body
+
+    @property
+    def log_headers(self):
+        return self._log_headers
+
+    @property
+    def headers_to_include(self):
+        return self._headers_to_include
+
+    @property
+    def headers_to_exclude(self):
+        return self._headers_to_exclude
+
+    @property
+    def headers_to_unmask(self):
+        return self._headers_to_unmask
+
+    def __init__(self, log_body, log_headers, headers_to_include,
```

Comment:
should we assign None default values here, can that cause any unfamiliar issue

Reply:
No, this can not cause any issues as default values are being handled through the SDK classes.

Reply:
yes we discussed that